### PR TITLE
Gfx12 - ocp fp8/bf8 support

### PR DESCRIPTION
**Pull request by cmingch**

### CODE REVIEW

**Comment by cmingch** [comment]
need rocm build > 14631
gfx12
[----------] Global test environment tear-down
[==========] 22565 tests from 13 test suites ran. (1308907 ms total)
[  PASSED  ] 22565 tests.
hipBLASLt version: 1000

py3: exit 0 (3830.68 seconds) /workspace/hipBLASLt/tensilelite> py.test -v --basetemp=/tmp/.tensile-tox/py3/tmp --junit-xml=/workspace/hipBLASLt/tensilelite/python_tests.xml --junit-prefix=py3 --color=yes -n 4 --prebuilt-client=/tmp/.tensile-tox/py3/client/0_Build/client/tensile_client Tensile/Tests -m common --junit-xml=build/python_unit_tests.xml --timing-file=build/timing.csv pid=3275155
  py3: OK (3959.74=setup[7.20]+cmd[121.86,3830.68] seconds)
  congratulations :) (3959.78 seconds)


gfx942

[----------] Global test environment tear-down
[==========] 35886 tests from 13 test suites ran. (3449272 ms total)
[  PASSED  ] 35886 tests.
hipBLASLt version: 1000

gfx11
py3: exit 0 (1927.22 seconds) /workspace/hipBLASLt/tensilelite> py.test -v --basetemp=/tmp/.tensile-tox/py3/tmp --junit-xml=/workspace/hipBLASLt/tensilelite/python_tests.xml --junit-prefix=py3 --color=yes -n 4 --prebuilt-client=/tmp/.tensile-tox/py3/client/0_Build/client/tensile_client Tensile/Tests -m common --junit-xml=build/python_unit_tests.xml --timing-file=build/timing.csv pid=106127
  py3: OK (2111.45=setup[10.90]+cmd[173.32,1927.22] seconds)
  congratulations :) (2111.50 seconds)

**Review by KKyang — DISMISSED** [review]

File: clients/include/type_dispatch.hpp [hunk 1]
```diff
@@ -181,6 +246,46 @@ auto hipblaslt_matmul_dispatch(const Arguments& arg)
         {
             return TEST<hipblaslt_bf8_fnuz, hipblaslt_bf8_fnuz, hipblasLtBfloat16, float>{}(arg);
         }
+        else if(TiA == HIP_R_8F_E4M3 && TiB == HIP_R_8F_E4M3 && To == HIP_R_32F
```
Comment (TonyYHsieh) [inline clients/include/type_dispatch.hpp]: should we add #ifdef ROCM_USE_FLOAT8 in here??

Comment (cmingch) [reply clients/include/type_dispatch.hpp]: done

**Review by TonyYHsieh — APPROVED** [review]
LGTM

**Review by KKyang — APPROVED** [review]

**Pull request merged**